src/Components/allocations/helper_functions/handleSaveAllocation.ts

```diff
@@ -0,0 +1,27 @@
+import { Dispatch, MouseEvent, SetStateAction } from "react";
+import { errorType } from "../../../types";
+import { saveAllocationIntoDb } from "./saveAllocationIntoDb";
+
+type Props = {
+  event: React.FormEvent<HTMLFormElement>,
+  allocationName: string,
+  setReRender: Dispatch<SetStateAction<number>>;
+  handleClose: any,
+  setError: Dispatch<SetStateAction<errorType>>;
+}
+
+const handleSaveAllocation = ({ event, allocationName, setReRender, handleClose, setError }: Props) => {
```

Comment:
Same. You don't win anything "hiding" all the ugly stuff in one function. This function is bloated with responsibility (you can usually see it by the amount of its parameters).